src/main/java/net/fabricmc/loom/util/fmj/FabricModJsonFactory.java

```diff
@@ -109,7 +111,10 @@ public static Optional<FabricModJson> createFromZipOptional(Path zipPath) {
 
 	@Nullable
 	public static FabricModJson createFromSourceSetsNullable(Project project, SourceSet... sourceSets) throws IOException {
-		final File file = SourceSetHelper.findFirstFileInResource(FABRIC_MOD_JSON, project, sourceSets);
+		final RegularFileProperty fmjPath = project.getExtensions().getByType(LoomGradleExtensionAPI.class).getFabricModJsonPath();
```

Comment:
I dont think this right place to do this, `FabricModJsonHelpers.getModsInProject` is the place for it.

Reply:
What exactly? My implementation was based on the fact that the `createFromSourceSetsNullable` method is called from the Loom and Fabric API extensions (`createFromSourceSetsNullable.getModVersion` and `FabricApiAbstractSourceSet.configureSourceSet`), so using the property here would provide the desired file to those too.